tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,335 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Managing Virtual Networks with virsh included
+# the following priorities 4 types of virtual network:
+# P1 HOST bridge network
+# P2 NAT based network
+# P3 Routed network
+# P4 Isolated network
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use set_config_as_glue;
+use version_utils 'is_sle';
+
+
+our $virt_host_bridge = 'br0';
+sub prepare_network {
+    my $config_path = "/etc/sysconfig/network/ifcfg-$virt_host_bridge";
+    if (script_run("[[ -f $config_path ]]") != 0) {
+        assert_script_run("ip link add name $virt_host_bridge type bridge");
+        assert_script_run("ip link set dev $virt_host_bridge up");
+        my $bridge_setup = <<'EOF';
+#!/bin/bash
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='none'\nSTARTMODE='auto'"
+bridge="BOOTPROTO='dhcp'\nBRIDGE='yes'\nBRIDGE_FORWARDDELAY='0'\nBRIDGE_PORTS='$ACTIVE_NET'\nBRIDGE_STP='off'\nSTARTMODE='auto'"
+> /etc/sysconfig/network/ifcfg-br0.new
+echo -e $bridge >/etc/sysconfig/network/ifcfg-br0
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-br0
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+systemctl restart network.service;
+EOF
+        script_output($bridge_setup, 900);
+    }
+}
+
+sub restore_standalone {
+    my $standalone_path = "/usr/share/qa/qa_test_virtualization/shared/standalone";
+    if (script_run("[[ -f $standalone_path ]]") == 0) { assert_script_run("source $standalone_path", 60); }
+}
+
+sub destroy_standalone {
+    my $cleanup_path = "/usr/share/qa/qa_test_virtualization/cleanup";
+    if (script_run("[[ -f $cleanup_path ]]") == 0) { assert_script_run("source $cleanup_path", 60); }
```

Comment:
`assert_script_run("source $cleanup_path", 60) if (script_run("[[ -f $cleanup_path ]]") == 0);` should be better

Reply:
Great! Had updated code as your mentioned.